src/main/scala/tech/cryptonomic/conseil/util/RouteUtil.scala

```diff
@@ -0,0 +1,112 @@
+package tech.cryptonomic.conseil.util
+
+import java.util.UUID
+
+import akka.http.scaladsl.model.StatusCodes.InternalServerError
+import akka.http.scaladsl.model.{HttpEntity, HttpResponse, RemoteAddress, StatusCodes}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.directives.BasicDirectives
+import akka.http.scaladsl.server.{Directive, ExceptionHandler, RequestContext, Route}
+import akka.stream.Materializer
+import cats.effect.concurrent.MVar
+import cats.effect.{Concurrent, IO}
+import com.typesafe.scalalogging.LazyLogging
+import org.slf4j.MDC
+
+/** Utility class for recording responses */
+class RouteUtil(implicit concurrent: Concurrent[IO]) extends LazyLogging {
+
+  private val requestInfoMap: MVar[IO, Map[UUID, RequestValues]] =
+    MVar.of[IO, Map[UUID, RequestValues]](Map.empty[UUID, RequestValues]).unsafeRunSync()
+
+  /** Directive adding recorded values to the MDC */
+  def recordResponseValues(
+      ip: RemoteAddress
+  )(implicit materializer: Materializer, correlationId: UUID): Directive[Unit] =
+    BasicDirectives.extractRequestContext.flatMap { ctx =>
```

Comment:
I guess we could simply `extractRequest` since that's what the context is used for.